RevenueCatUI/Templates/V2/Components/Carousel/CarouselComponentView.swift

```diff
@@ -0,0 +1,693 @@
+        if index < originalCount {
+            let newCopyIndex = lowestCopyIndex() - 1
+            let newItems = makeItems(forCopyIndex: newCopyIndex)
+            data.insert(contentsOf: newItems, at: 0)
+            index += originalCount // keep user in same “visual” position
+        }
+
+        // If index is in the last copy, append another
+        if index >= data.count - originalCount {
+            let newCopyIndex = highestCopyIndex() + 1
+            let newItems = makeItems(forCopyIndex: newCopyIndex)
+            data.append(contentsOf: newItems)
+        }
+    }
+
+    // MARK: - Pruning
+
+    private func pruneDataIfNeeded() {
+        let copiesInData = data.count / originalCount
+        let maxCopiesAllowed = 5
+
+        guard copiesInData > maxCopiesAllowed else { return }
+
+        // If user is at least 2 copies in from the front, we can drop 1 copy from the front.
+        while index >= 2 * originalCount,
+              data.count / originalCount > maxCopiesAllowed {
+            data.removeFirst(originalCount)
+            index -= originalCount
+        }
+
+        // If user is at least 2 copies from the end, we can drop 1 copy from the end.
+        while index < data.count - 2 * originalCount,
+              data.count / originalCount > maxCopiesAllowed {
+            data.removeLast(originalCount)
+        }
+    }
+
+    // Identify which copy indices we have
+    private func lowestCopyIndex() -> Int {
+        guard let firstID = data.first?.id else { return 0 }
+        return firstID / originalCount
+    }
+
+    private func highestCopyIndex() -> Int {
+        guard let lastID = data.last?.id else { return 0 }
+        return lastID / originalCount
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct PageControlView: View {
+
+    let originalCount: Int
+    let pageControl: DisplayablePageControl
+    @Binding var currentIndex: Int
+
+    @State private var localCurrentIndex: Int = 0
+
+    var activeIndicator: DisplayablePageControlIndicator {
+        pageControl.active
+    }
+
+    var indicator: DisplayablePageControlIndicator {
+        pageControl.default
+    }
+
+    var body: some View {
+        if self.originalCount <= 1 {
+            EmptyView()
+        } else {
+            HStack(spacing: self.pageControl.spacing) {
+                ForEach(0..<originalCount, id: \.self) { index in
+                    Capsule()
+                        .fill(localCurrentIndex == index ? activeIndicator.color : indicator.color)
+                        .frame(
+                            width: localCurrentIndex == index ? activeIndicator.width : indicator.width,
+                            height: localCurrentIndex == index ? activeIndicator.height : indicator.height
+                        )
+                        .animation(.easeInOut, value: self.localCurrentIndex)
+                }
+            }
+            .padding(self.pageControl.padding)
```

Comment:
Would it make sense here to ignore the horizontal padding? I say this to avoid these kinds of clipping of the cards on the edges. But I don't know if ignoring these paddings could have other undesired consequences
![image](https://github.com/user-attachments/assets/a45b5c40-ce2c-4a14-81c2-9539de9206e6)

Reply:
Hmm it's a good point... But for now, I think we should just limit setting it in the dashboard.